### DISCUSSION TITLE
Improving the quality of the generated PDF files

### DISCUSSION THREAD

**Discussion by damithc** [MarkBind/markbind#2074] [Ideas]
One area we have room to improve (and relevant to our target market) is the quality of the generated PDF file.
For example, when students generate a PDF file from the textbook that is a MarkBind website. In particular, optimizing the layout w.r.t. page breaks.

Here is an example textbook page and the PDF.
https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/print.html
[textbook.pdf](https://github.com/MarkBind/markbind/files/10419445/textbook.pdf)

**Comment by tlylt** [comment]
Some resources
- Invisible newlines that only show up in print can be added via [the `d-none d-print-block` class ](https://getbootstrap.com/docs/4.0/utilities/display/#display-in-print)
- page break for print might be adjustable via [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/page-break-before)

**Reply by damithc** [reply]
> * Invisible newlines that only show up in print can be added via [the `d-none d-print-block` class](https://getbootstrap.com/docs/4.0/utilities/display/#display-in-print)

Manually adding such hidden breaks might not be convenient as we wouldn't know where exactly the page break would happen, and those places can change as the content evolve. 

In MS Word, you can configure headings of a specific level to 'keep with next' so that they will not be stranded at the bottom of a page. I wonder something similar can be done for MarkBind.
![image](https://user-images.githubusercontent.com/1673303/212523463-ddfe4e03-a8d5-41b0-b261-c2e66914b84c.png)

**Reply by tlylt** [reply]
Agree on the content evolution point. For simple webpages, users can use the print preview to review the page breaks, though not very convenient.
